server/lib/ReportManager.js

```diff
@@ -97,6 +98,13 @@ class ReportManager {
       throw new BadRequest(`The application does not support rendering ${options.renderer}.`, 'ERRORS.INVALID_RENDERER');
     }
 
+    // @TODO user information could be determined by report manager, removing the need for this check
+    if (this.options.saveReport && !this.options.user) {
+      let invalidSaveDescription = 'Report cannot be saved without providing a `user` entity to ReportManager';
+      console.error(invalidSaveDescription);
+      throw new InternalServerError(invalidSaveDescription);
```

Comment:
This looks fine to me in terms of code.

I would encourage that instead of `console`, we use `winston` so that we can control the type and destination of all logged text.  It doesn't need to be done in the PR, as the whole server will need to be reviewed so that errors are properly logged, and useful information is logged.